### DISCUSSION TITLE
ES import Service from 'node-windows'

### DISCUSSION THREAD

**Discussion by Harvey-Mushman** [coreybutler/node-windows#357] [Q&A] [answered]
I'm not sure if I fully understand the difference between 

const Service = require('node-windows').Service

which is documented in the readme and the newer es6 syntax using import.  So after trial and several errors, I managed to re-write the sample code to read as follows:

import nodeWindows from 'node-windows'
const Server = nodeWindows.Server

This worked without throwing an es6 error and after looking at Task Manager, I was able to confirm my server is now running.

Does anyone with better understanding of the syntax want to confirm or correct my understanding?

**Comment by miquelfire** [comment, marked as answer]
I think you also could do this (untested, there might be a special syntax CommonJS libraries need to be written in to allow this):
```javascript
import { Service } from 'node-windows';
```

Note: When this library was written (and the maintainer had time to keep things up to date), ESM wasn't a thing yet, which is why none of the documentation talks about that format.

**Reply by coreybutler** [reply]
This comment is spot on.

I tried to create a version that's more ES6 friendly some time ago, but ran out of time. I still have a desire to do this when time allows, but I don't know when that will be. I want to replace winsw with a custom Go app designed specifically to interact with JS runtime code.
